architectures/firmware/index.rst

```diff
@@ -6,14 +6,7 @@ Firmware Architecture
 .. toctree::
    :maxdepth: 1
 
-   overview
-   mem-mgmt
-   pm-runtime/index
-   schedulers
-   drivers/index
-   components/index
-   pipelines/index
-   kd_integration/index
+   sof-legacy/index
```

Comment:
Probably better to say sof-xtos instead of legacy as xtos still used. Fwiw, I intend to do the same name fix in FW for v2.4.

Reply:
@lgirdwood legacy naming has been updated with xtos